sound/soc/sdca/sdca_functions.c

```diff
@@ -1222,6 +1222,124 @@ static int find_sdca_entity_ge(struct device *dev,
 	return -EINVAL;
 }
 
+static int sdwhid_parse(struct hid_device *hid)
+{
+	//struct sdw_slave *slave = dev_to_sdw_dev(hid->dev.parent);
+	struct sdca_entity *entity = hid->driver_data;
+	unsigned int rsize = 0;
+	char *rdesc;
+	int ret;
+
+	rsize = entity->hide.hid_desc.desc[0].wDescriptorLength;
+
+	if (!rsize || rsize > HID_MAX_DESCRIPTOR_SIZE) {
+		dev_err(&hid->dev, "weird size of report descriptor (%u)\n", rsize);
+		return -EINVAL;
+	}
+
+	rdesc = kmalloc(rsize, GFP_KERNEL);
+	if (!rdesc)
+		return -ENOMEM;
+
+	memcpy(rdesc, entity->hide.hid_report_desc, rsize);
+
+	ret = hid_parse_report(hid, rdesc, rsize);
+	kfree(rdesc);
+	if (ret) {
+		dev_err(&hid->dev, "parsing report descriptor failed\n");
+		goto err;
+	}
+
+	return 0;
+err:
+	return ret;
+}
+
+static int sdwhid_start(struct hid_device *hid)
+{
+	return 0;
```

Comment:
these are mandatory and we don't need to do anything?

Reply:
Please check the comment above.